### PR TITLE
Misc trivial typo fixes

### PR DESCRIPTION
**Pull request by luzpaz**
Many user-facing (incl. doxy) + many that are source code typos. Found using `codespell -q 3 --skip="*.po,./src/external" -I ../darktable-whitelist.txt`
where whitelist consisted of:
```
cas
childs
ect
eacg
isnt
liquify
nd
nonexistant
russian
substract
thru
```

### CODE REVIEW

File: src/develop/blend.h [hunk 1]
```diff
@@ -374,7 +374,7 @@ gboolean dt_develop_blend_params_is_all_zero(const void *params, size_t length);
 /** update blendop params from older versions */
 int dt_develop_blend_legacy_params(dt_iop_module_t *module, const void *const old_params,
                                    const int old_version, void *new_params, const int new_version,
-                                   const int lenght);
+                                   const int length);
```
Comment (luzpaz) [inline src/develop/blend.h]: please review this revision

**Comment by houz** [comment]
Sorry, we are in string freeze right now. If you split this up we would accept all changes that are either source/comments or user manual.

File: src/develop/blend.h (hunk 1 above)
Comment (luzpaz) [reply src/develop/blend.h]: @houz is this a correct revision ? or shall i exclude it?

**Comment by luzpaz** [comment]
> Sorry, we are in string freeze right now. If you split this up we would accept all changes that are either source/comments or user manual.

I will split up the PR

**Comment by luzpaz** [comment]
Closing PR in favor of #1577 and #1578

**Pull request closed without merging**